Firestore/core/src/util/hashing.h

```diff
@@ -199,6 +201,12 @@ size_t RankedInvokeHash(K value, HashChoice<5>) {
   return InvokeHash(underlying);
 }
 
+template <typename K>
+auto RankedInvokeHash(const std::unique_ptr<K>& ptr, HashChoice<6>)
+    -> decltype(InvokeHash(*ptr)) {
+  return ptr ? InvokeHash(*ptr) : 23631;
+}
+
```

Comment:
Will this code ever get executed?

Based on the comments above:

```
 * HashChoice is a recursive type, defined such that HashChoice<0> is the most
 * specific type with HashChoice<1> and beyond being progressively less
 * specific. This causes the compiler to prioritize the overloads with
 * lower-numbered HashChoice types, allowing compilation to succeed even if
 * multiple specializations match.
 ```

And `HachChoice<2>`: `Hashes the given value if it has a specialization of std::hash.`

And `std::hash` does have a specialization for `std::unique_ptr`:
https://en.cppreference.com/w/cpp/memory/unique_ptr/hash

Reply:
Nope, because all these settings object do not support std::hash, we use our own Hash() functions.

Reply:
Oh I see... Thanks for the clarification